tests/engine/utility_test.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+from scripts.engine.core.constants import Shape
+from scripts.engine import utility
+
+
+class TestUtility:
+
+    test_get_coordinates_from_shape_parameters = [
+        (Shape.SQUARE, 1, (0, 0), [(-1, -1), (-1, 0), (-1, 1), (0, -1), (0, 0), (0, 1), (1, -1), (1, 0), (1, 1)]),
+        (Shape.CIRCLE, 1, (0, 0), [(-1, -1), (-1, 0), (-1, 1), (0, -1), (0, 0), (0, 1), (1, -1), (1, 0), (1, 1)]),
+        (Shape.CROSS, 1, (0, 0), [(0, 0), (1, -1), (-1, 1), (-1, -1), (1, 1)]),
+        (Shape.TARGET, 1, (0, 0), [(0, 0)]),
+        (Shape.CONE, 1, (0, 1), [(0, 0), (1, 1), (0, 1), (-1, 1)]),
+        (Shape.CONE, 1, (0, -1), [(0, 0), (1, -1), (0, -1), (-1, -1)]),
+        (Shape.CONE, 1, (1, 0), [(0, 0), (1, -1), (1, 1), (1, 0)]),
+        (Shape.CONE, 1, (-1, 0), [(0, 0), (-1, 1), (-1, 0), (-1, -1)]),
+    ]
+
+    @pytest.mark.parametrize("shape, size, direction, expected", test_get_coordinates_from_shape_parameters)
```

Comment:
To avoid repeting code, parametrize it so pytest can generate all the combinations we want to test for.